### DISCUSSION TITLE
Exception Snapshot for Visual Studio

### DISCUSSION THREAD

**Discussion by D2O84** [dotnet/csharplang#5332] [General] [answered]
Would it be possible to save the current state as a Visual Studio debugger snapshot in a file, 
Such as having the right version of the project open and opening the debugger snapshot file in VS we would be able to have full Call Stack (and the ability to click any level in that call stack to change context), Local Window, Watch Window, Threads, Tasks, ... as is you had hit a breakpoint while debugging locally, I understand it would be impossible to resume execution, but at least seeing the context at different level of the call stack would help.

This would be great for production, especially if we can generate such file whenever an exception is raised. It would make it easier to identify errors on the caller side.

In addition, it would be great to store them in different folders, or file names with specific prefix based on the stack trace so they are grouped by cases (i.e. the exceptions generate in similar context are together, in the sense that if you fix one of those issues you'll probably fix all the issues in that group)

```
<configuration>
  <system.diagnostics>
      <exception>
          <add name="..." type="System.Diagnostics.ExceptionSnapshotRecorder" initializeData="../logs/exceptions"/>
      </exception>
  </system.diagnostics>
</configuration>
```

and

`System.Diagnostics.SnapshotRecorder.Record(@$"../logs/event/{DateTime.Now.ToString("yyyy_MM_dd_HH_mm_ss").txt");`

Thank you

Transfered to
#https://github.com/dotnet/runtime/discussions/60805

**Comment by HaloFour** [comment, marked as answer]
This sounds like a runtime tooling request and not a change for the C# language.  Maybe try the [runtime](https://github.com/dotnet/runtime/) repo instead?

**Reply by D2O84** [reply]
https://github.com/dotnet/runtime/discussions/60805
